db/background_mgr_resync_dcp_test.go

```diff
@@ -444,6 +445,93 @@ function sync(doc, oldDoc){
 	return db, ctx
 }
 
+// TestResyncMou ensures that resync updates create mou, and preserve pcas in mou in the case where resync is reprocessing an import
+func TestResyncMou(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Test requires Couchbase Server")
+	}
+
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyMigrate, base.KeyImport)
+	db, ctx := setupTestDBWithOptionsAndImport(t, nil, DatabaseContextOptions{})
+	defer db.Close(ctx)
+
+	if !db.Bucket.IsSupported(sgbucket.BucketStoreFeatureMultiXattrSubdocOperations) {
+		t.Skip("Test requires multi-xattr subdoc operations, CBS 7.6 or higher")
+	}
+
+	db.Options.QueryPaginationLimit = 100 // Required for principal ID query to not deadlock
```

Comment:
It's like the comment says - otherwise the principal ID query will deadlock when QueryPaginationLimit is zero, and we normally initialize this outside the db package.  Possibly worth cleanup, but not in this PR I don't think.